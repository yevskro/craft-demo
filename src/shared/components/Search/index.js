import React from 'react';
import styled from 'styled-components';

function Search({ search, searchCbFn }) {
  console.log(search);
  return (
    <SearchContainer>
      <SearchIcon src="/search.png" />
      <SearchInput
        type="text"
        onChange={(e) => searchCbFn(e)}
        value={search}
        placeholder="search"
      />
    </SearchContainer>
  );
}

const SearchContainer = styled.div`
  display: flex;
  justify-content: center;
  margin-top: 25px;
`;

const SearchIcon = styled.img`
  width: 12px;
  height: 12px;
  transform: translate(17px, 5px);
`;

const SearchInput = styled.input`
  border-radius: 20px;
  border-color: black;
  padding-left: 20px;
`;

export default Search;
