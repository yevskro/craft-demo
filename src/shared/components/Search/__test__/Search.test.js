import React from 'react';
import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import Search from '../index';

test('renders without crashing', () => {
  render(<Search />);
});

test('renders search with placeholder', () => {
  render(<Search />);
  expect(screen.getByPlaceholderText('search')).toBeInTheDocument();
});

test('renders search with search text', () => {
  render(<Search search="test" />);
  expect(screen.getByDisplayValue('test')).toBeInTheDocument();
});

test('invokes onSearch when typing search', () => {
  const onSearchMock = jest.fn();

  render(<Search search="" onSearch={onSearchMock} />);

  userEvent.type(screen.getByPlaceholderText('search'), 'test');
  expect(onSearchMock).toHaveBeenCalledTimes(4);
});
