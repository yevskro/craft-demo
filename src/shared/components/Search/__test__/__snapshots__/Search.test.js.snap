// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  margin-top: 25px;
}

.c1 {
  width: 12px;
  height: 12px;
  -webkit-transform: translate(17px,5px);
  -ms-transform: translate(17px,5px);
  transform: translate(17px,5px);
}

.c2 {
  border-radius: 20px;
  border-color: black;
  padding-left: 20px;
}

<div
  className="c0"
>
  <img
    alt="search icon"
    className="c1"
    src="/search.png"
  />
  <input
    className="c2"
    onChange={[Function]}
    placeholder="search"
    type="text"
    value=""
  />
</div>
`;
